File: problems/Python/45. Jump Game II.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        jmp = 0
        end = 0
        far = 0
        
        for i in range(len(nums) - 1):
            far = max(far, i + nums[i])
            
            if i == end:
                jmp += 1
                end = far
                
        return jmp

File: problems/C++/45. Jump Game II.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int jmp = 0;
        int end = 0;
        int far = 0;

        for (int i = 0; i < nums.size() - 1; ++i) {
            far = max(far, i + nums[i]);

            if (i == end) {
                jmp++;
                end = far;
            }
        }
        return jmp;
    }
};
